src/org/omegat/core/machinetranslators/DeepLTranslate.java

```diff
@@ -147,22 +148,17 @@ private String unescapeHTML(String text) {
 
     @SuppressWarnings("unchecked")
     protected String getJsonResults(String json) {
-        Map<String, Object> rootNode;
+        JsonNode rootNode;
+        ObjectMapper mapper = new ObjectMapper();
         try {
-            rootNode = (Map<String, Object>) JsonParser.parse(json);
+            rootNode = mapper.readTree(json);
         } catch (Exception e) {
             Log.logErrorRB(e, "MT_JSON_ERROR");
             return OStrings.getString("MT_JSON_ERROR");
         }
 
         // { "translations": [ { "detected_source_language": "DE", "text": "Hello World!" } ] }
-        try {
-            List<Object> translationsList = (List<Object>) rootNode.get("translations");
-            Map<String, String> translationNode = (Map<String, String>) translationsList.get(0);
-            return translationNode.get("text");
-        } catch (NullPointerException e) {
-            return null;
-        }
+        return rootNode.get("translations").get(0).get("text").asText();
```

Comment:
The old code allowed (in a dumb way) for something to be null. How does the new code behave if there is a null?

Reply:
(I don't know anything about Jackson but I assume this blows up with an NPE)

Reply:
Updated.